test/server/config/network/config_test.cc

```diff
@@ -338,6 +338,29 @@ TEST(AccessLogConfigTest, FileAccessLogTest) {
   EXPECT_NE(nullptr, dynamic_cast<AccessLog::FileAccessLog*>(instance.get()));
 }
 
+// Test that a minimal TcpProxy v2 config works.
+TEST(TcpProxyConfigTest, TcpProxyConfigTest) {
+  auto factory = Registry::FactoryRegistry<NamedNetworkFilterConfigFactory>::getFactory(
+      Config::NetworkFilterNames::get().TCP_PROXY);
+  ASSERT_NE(nullptr, factory);
+
+  ProtobufTypes::MessagePtr message = factory->createEmptyConfigProto();
```

Comment:
nit: This is the pattern we have been using to do this type of test: https://github.com/ggreenway/envoy/blob/23faab6835ababc58a80c407c01c273ecf82e090/test/server/config/network/mongo_proxy_test.cc#L53. The dynamic_cast tests that it is the right thing and then you fill it.